tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/passes/ConstraintGenPassImpl.scala

```diff
@@ -0,0 +1,48 @@
+package at.forsyte.apalache.tla.bmcmt.passes
+
+import at.forsyte.apalache.infra.passes.{Pass, PassOptions}
+import at.forsyte.apalache.tla.bmcmt._
+import at.forsyte.apalache.tla.lir.NullEx
+import at.forsyte.apalache.tla.lir.transformations.{LanguagePred, LanguageWatchdog}
+import com.google.inject.Inject
+import com.google.inject.name.Named
+import com.typesafe.scalalogging.LazyLogging
+
+/**
+ * The implementation of a bounded model checker with SMT.
+ *
+ * @author Jure Kukovec
+ */
```

Comment:
This comment is confusing. Our main model checker is also a bounded model checker with SMT. What you are doing here is a translator from a restricted fragment to VMT. It may evolve into a complete model checker later. But this needs a description.

Reply:
Also a copy-paste artifact.